File: rb/features.rb
# Microlink SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module MicrolinkFeatures
  def self.make_feature(name)
    case name
    when "base"
      MicrolinkBaseFeature.new
    when "test"
      MicrolinkTestFeature.new
    else
      MicrolinkBaseFeature.new
    end
  end
end

File: py/microlink_sdk/features.py
# Microlink SDK feature factory

from microlink_sdk.feature.base_feature import MicrolinkBaseFeature
from microlink_sdk.feature.test_feature import MicrolinkTestFeature


def _make_feature(name):
    features = {
        "base": lambda: MicrolinkBaseFeature(),
        "test": lambda: MicrolinkTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
